metrics.go

```diff
@@ -126,8 +126,18 @@ func (se *statsExporter) metricToMpbTs(ctx context.Context, metric *metricdata.M
 	if metric == nil {
 		return nil, errNilMetricOrMetricDescriptor
 	}
-
-	resource := se.metricRscToMpbRsc(metric.Resource)
+	var resource *monitoredrespb.MonitoredResource
+	if get := se.o.ResourceByDescriptor; get != nil {
+		mr := get(&metric.Descriptor)
```

Comment:
What `metric.Descriptor` will be if I register the following:

```
requestCountM = stats.Int64(
       "request_count",
	"The number of requests",
	stats.UnitDimensionless)
view.Register(
		&view.View{
			Description: "The number of requests",
			Measure:     requestCountM,
			Aggregation: view.Count(),
			TagKeys: append(metrics.CommonRevisionKeys, metrics.PodTagKey, metrics.ContainerTagKey,
				metrics.ResponseCodeKey, metrics.ResponseCodeClassKey, metrics.NumTriesKey),
		})
```

Reply:
```go
metric.Descriptor{
	Name:        "request_count",
	Description: "The number of requests",
	Unit:        metricdata.	UnitDimensionless,
	Type:        metric.TypeCumulativeInt64,
}
```

Reply:
Could we determine resource type by name? Name is more like an ID instead of the description. WDYT?

Reply:
actually Descriptor.Name = View.Name.

knative does determine resource type from name as per [this](https://github.com/knative/pkg/blob/f684a1078ac309f4b32e6240d9b622bea092296e/metrics/stackdriver_exporter.go#L130).
It seems you also need tags, can you confirm?

Reply:
Ah got it.

The tags are used to set the values of monitored resource labels. For example, `project_id`.